src/bootstrap/config.rs

```diff
@@ -95,7 +97,8 @@ pub struct Config {
     pub llvm_release_debuginfo: bool,
     pub llvm_version_check: bool,
     pub llvm_static_stdcpp: bool,
-    pub llvm_link_shared: bool,
+    /// `None` if `llvm_from_ci` is true and we haven't yet downloaded llvm.
+    pub llvm_link_shared: Cell<Option<bool>>,
```

Comment:
Can we add a comment here to use this via the wrapper function, and perhaps either try to refactor to make this private or rename it to something like llvm_link_shared_internal or some similar longer & somewhat evocative of don't use me name?

Reply:
Yeah, so I actually tried to make it private at first, but it broke the tests:
```
error[E0451]: field `llvm_link_shared` of struct `config::Config` is private
  --> src/bootstrap/builder/tests.rs:67:43
   |
67 |         let config = Config { stage: 0, ..configure("build", &["A"], &["A"]) };
   |                                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ field `llvm_link_shared` is private
```
I could either do a hacky
```rust
    /// `None` if `llvm_from_ci` is true and we haven't yet downloaded llvm.
    #[cfg(not(test))]
    llvm_link_shared: Cell<Option<bool>>,
    #[cfg(test)]
    pub llvm_link_shared: Cell<Option<bool>>,
```
or add `_internal` like you suggested.

Reply:
That seems fine for now.